group_theory/subgroup.lean

```diff
@@ -269,3 +270,160 @@ dvd.intro (@fintype.card (cosets (gpowers a)) ft_cosets) $
 end classical
 
 end
+
+
+variables {G : Type u} {H : Type v}
+
+class normal_subgroup [group G] (S : set G) extends is_subgroup S : Prop :=
+(normal : ∀ n ∈ S, ∀ g : G, g * n * g⁻¹ ∈ S)
+             
+namespace is_subgroup
+variable [group G]
+
+-- Normal subgroup properties
+lemma mem_norm_comm {a b : G} {S : set G} [normal_subgroup S] (hab : a * b ∈ S) : b * a ∈ S := 
+have h : a⁻¹ * (a * b) * a⁻¹⁻¹ ∈ S, from normal_subgroup.normal (a * b) hab a⁻¹,
+by simp at h; exact h
+
+-- Examples of subgroups
+def trivial (G : Type u) [group G] : set G := {1}
+
+lemma eq_one_of_trivial_mem {G : Type u} [group G] {g : G} (p : g ∈ trivial G) : g = 1 :=
+begin
+  cases p, 
+  assumption, 
+  cases p,
+end
+lemma trivial_mem_of_eq_one {G : Type u} [group G] {g : G} (p : g = 1) : g ∈ trivial G :=
+begin
+  rw p,
+  unfold trivial,
+  simp,
+end
+
+instance trivial_normal : normal_subgroup (trivial G) :=
+by refine {..}; simp [trivial] {contextual := tt}
+
+instance univ_subgroup : normal_subgroup (@univ G) :=
+by refine {..}; simp
+
+def center (G : Type u) [group G] : set G := {z | ∀ g, g * z = z * g}
+
+instance center_normal : normal_subgroup (center G) := {
+  one_mem := by simp [center],
+  mul_mem := begin
```

Comment:
`begin` should be at the next line, and the list of tactic calls inside indented by 2.
And everywhere further down too.

Reply:
Fixed!